chains.go

```diff
@@ -24,17 +24,25 @@ import (
 
 // AssembleServerChains takes in an array of certificates, finds all certificates with
 // x509.ExtKeyUsageAny or x509.ExtKeyUsageServerAuth and builds an array of leaf-first
-// chains.
-func AssembleServerChains(certs []*x509.Certificate) ([][]*x509.Certificate, error) {
+// chains. Chains are built starting from server authentication certificates found in `certs`
+// and the signer chains are built from `certs` and `cas`. Both slices are de-duped
+// and the `cas` slice is filtered for certificates with the CA flag set.
+func AssembleServerChains(certs []*x509.Certificate, cas []*x509.Certificate) ([][]*x509.Certificate, error) {
 	if len(certs) == 0 {
 		return nil, nil
 	}
 
+	certs = getUniqueCerts(certs)
+
 	var chains [][]*x509.Certificate
 
 	var serverCerts []*x509.Certificate
 
 	for _, cert := range certs {
+		//if we find CAs add them to the known CAs
+		if cert.IsCA {
```

Comment:
feels like 43's 'if' and the if(!cert.IsCA) on line 48 could be condensed for additional clarity.